Mathlib/AlgebraicGeometry/GammaSpecAdjunction.lean

```diff
@@ -0,0 +1,530 @@
+    toOpen (Γ.obj (op X)) (basicOpen r) ≫ X.toΓSpecSheafedSpace.c.app (op (basicOpen r)) ≫ f =
+      X.toToΓSpecMapBasicOpen r ≫ f := by
+  rw [←Category.assoc, toΓSpecSheafedSpace_app_spec]
+
+/-- The map on stalks induced by the unit commutes with maps from `Γ(X)` to
+    stalks (in `Spec Γ(X)` and in `X`). -/
+theorem toStalk_stalkMap_to_Γ_Spec (x : X) :
+    toStalk _ _ ≫ PresheafedSpace.stalkMap X.toΓSpecSheafedSpace x = X.ΓToStalk x := by
+  rw [PresheafedSpace.stalkMap]
+  erw [← toOpen_germ _ (basicOpen (1 : Γ.obj (op X)))
+      ⟨X.toΓSpecFun x, by rw [basicOpen_one] ; trivial⟩]
+  rw [← Category.assoc, Category.assoc (toOpen _ _)]
+  erw [stalkFunctor_map_germ]
+  -- Porting note : was `rw [←assoc, toΓSpecSheafedSpace_app_spec]`, but Lean did not like it.
+  rw [toΓSpecSheafedSpace_app_spec_comp]
+  unfold ΓToStalk
+  rw [← stalkPushforward_germ _ X.toΓSpecBase X.presheaf ⊤]
+  congr 1
+  change (X.toΓSpecBase _* X.presheaf).map le_top.hom.op ≫ _ = _
+  apply germ_res
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_stalk_stalk_map_to_Γ_Spec AlgebraicGeometry.LocallyRingedSpace.toStalk_stalkMap_to_Γ_Spec
+
+/-- The canonical morphism from `X` to the spectrum of its global sections. -/
+@[simps! val_base]
+def toΓSpec : X ⟶ Spec.locallyRingedSpaceObj (Γ.obj (op X)) where
+  val := X.toΓSpecSheafedSpace
+  prop := by
+    intro x
+    let p : PrimeSpectrum (Γ.obj (op X)) := X.toΓSpecFun x
+    constructor
+    -- show stalk map is local hom ↓
+    let S := (structureSheaf _).presheaf.stalk p
+    rintro (t : S) ht
+    obtain ⟨⟨r, s⟩, he⟩ := IsLocalization.surj p.asIdeal.primeCompl t
+    dsimp at he
+    set t' := _
+    change t * t' = _ at he
+    apply isUnit_of_mul_isUnit_left (y := t')
+    rw [he]
+    refine' IsLocalization.map_units S (⟨r, _⟩ : p.asIdeal.primeCompl)
+    apply (not_mem_prime_iff_unit_in_stalk _ _ _).mpr
+    rw [← toStalk_stalkMap_to_Γ_Spec]
+    erw [comp_apply, ← he]
+    rw [RingHom.map_mul]
+    -- Porting note : `IsLocalization.map_units` and the goal needs to be simplified before Lean
+    -- realize it is useful
+    have := IsLocalization.map_units (R := Γ.obj (op X)) S s
+    dsimp at this ⊢
+    refine ht.mul <| this.map _
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec AlgebraicGeometry.LocallyRingedSpace.toΓSpec
+
+theorem comp_ring_hom_ext {X : LocallyRingedSpace} {R : CommRingCat} {f : R ⟶ Γ.obj (op X)}
+    {β : X ⟶ Spec.locallyRingedSpaceObj R}
+    (w : X.toΓSpec.1.base ≫ (Spec.locallyRingedSpaceMap f).1.base = β.1.base)
+    (h :
+      ∀ r : R,
+        f ≫ X.presheaf.map (homOfLE le_top : (Opens.map β.1.base).obj (basicOpen r) ⟶ _).op =
+          toOpen R (basicOpen r) ≫ β.1.c.app (op (basicOpen r))) :
+    X.toΓSpec ≫ Spec.locallyRingedSpaceMap f = β := by
+  -- Porting note : `ext` did not pick `LocallyRingedSpace.Hom.ext` up
+  refine Hom.ext _ _ ?_
```

Comment:
It probably just needs an additional `@[ext]` lemma expressed in terms of the `\hom` for `LocallyRingedSpace`.

Reply:
I'm not sure I understand, `LocallyRingedSpace.Hom.ext` is already an `ext` lemma

Reply:
ah, you mean `⟶`, I am stupid